contrib/devtools/github-merge.py

```diff
@@ -47,17 +47,41 @@ def git_config_get(option, default=None):
     except subprocess.CalledProcessError:
         return default
 
-def retrieve_json(req, ghtoken):
+def get_response(req_url, ghtoken):
+    req = Request(req_url)
+    if ghtoken is not None:
+        req.add_header('Authorization', 'token ' + ghtoken)
+    return urlopen(req)
+
+def retrieve_json(req_url, ghtoken, use_pagination=False):
     '''
     Retrieve json from github.
     Return None if an error happens.
     '''
     try:
-        if ghtoken is not None:
-            req.add_header('Authorization', 'token ' + ghtoken)
-        result = urlopen(req)
         reader = codecs.getreader('utf-8')
-        obj = json.load(reader(result))
+        if not use_pagination:
+            return json.load(reader(get_response(req_url, ghtoken)))
+
+        obj = []
+        page_num = 1
+        while True:
+            req_url_page = '%s?page=%d' % (req_url, page_num)
```

Comment:
Style-nit: Could use str.format

```suggestion
            req_url_page = '{}?page={}'.format(req_url, page_num)
```